streams/core.py

```diff
@@ -602,3 +601,32 @@ def flush(self, _=None):
         out = tuple(self.cache)
         self.emit(out)
         self.cache.clear()
+
+
+def _stream_map(obj, func=None, **kwargs):
+    if hasattr(obj, '__stream_map__'):
```

Comment:
If I understand correctly, can we interchange using `__stream_map__` or the `singledispatch` implementation? Do you use `singledispatch` just to define a mapping without having to modify the object itself?